evm/chains/mainnet/__init__.py

```diff
@@ -56,13 +57,14 @@ def validate_header(cls, header, previous_header):
                 )
 
 
-class MainnetHomesteadVM(MainnetDAOValidatorVM, HomesteadVM):
-    pass
+class MainnetPostDAOHomesteadVM(MainnetDAOValidatorVM, HomesteadVM):
+    caused_chain_split = True
```

Comment:
I like that, but if we're going to special case the HomesteadVM, we might as well do it like this?
```
    if self.network_id != MAINNET_NETWORK_ID:
        continue
    dao_fork_block = HomesteadVM.dao_fork_block_number
    # request and validate dao_fork_block...
```

Then we don't have to worry about passing `chain_class` around as well

Reply:
Except that we don't know that we're using the `HomesteadVM` or whether it's a subclass which starts the fork block at a different height....  I think we have to pass in the `vm_configuration` in some form, otherwise this won't work for non-mainnet and similar VMs.